File: core/repository/service_lifecycle_engine.py
from __future__ import annotations

from typing import Any, Dict, List, Optional

from core.repository.service_runtime_graph_engine import build_service_runtime_graph


def model_service_lifecycle(source: str, path: str = "", files: Optional[List[str]] = None) -> Dict[str, Any]:
    g = build_service_runtime_graph(source, path, files)
    lifecycle = [{"service": n, "phase": "running"} for n in g.get("nodes", [])[:50]]
    return {"lifecycle": lifecycle, "evidence": g.get("evidence", [])}

File: src/repository/serviceLifecycleEngine.ts
/**
 * Converted from Python: core/repository/service_lifecycle_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { buildServiceRuntimeGraph } from "./serviceRuntimeGraphEngine.js";

export function modelServiceLifecycle(source: any, path: any = "", files: any = null): any {
  var g: any = buildServiceRuntimeGraph(source, path, files);
  var lifecycle: any = py.iter(py.slice(py.get(g, "nodes", []), null, 50)).map((n: any) => ({"service": n, "phase": "running"}));
  return {"lifecycle": lifecycle, "evidence": py.get(g, "evidence", [])};
}
export { buildServiceRuntimeGraph };
